src/Channel.js

```diff
@@ -1,54 +1,145 @@
-import {useState} from 'react'
+import {useState, useEffect} from 'react'
 import {supabase} from './supabaseClient'
 
 export default function Account({session}) {
 	const [loading, setLoading] = useState(false)
+	const [deleting, setDeleting] = useState(false)
 	const [name, setName] = useState('')
 	const [slug, setSlug] = useState('')
+	const [channel, setChannel] = useState('')
 
-	async function saveChannel({name, slug}) {
+	useEffect(() => {
+		findChannel()
+	}, [session])
+
+	async function findChannel() {
```

Comment:
maybe we could have these utils functions in a separate file/folder?
could start with `./utils/api.js`

Reply:
@jfalxa any idea on a pattern that would allow:
- async functions handling (to condition render of html elements or their content, such as loading state)
- reusable functions, that call the API through supabase sdk
- nice folder/file architecture

Reply:
yes, I use this pattern https://gist.github.com/jfalxa/4bfee2f37e911840d6f447526eeccf71

basically a hook that returns an tuple of 2 values:
`const [snapshot, startRequest] = useSnapshot(promiseBuilder, dependencies)`

The snapshot holds data about the promise loading state as well as errors and resolved value. All of it is stored in a react state so for every change, a rerender will happen, allowing easy update of the UI.

Reply:
Then I also have a file `api.ts` that has one function per endpoint in my backend api. The parameters of the function should match the parameters of the endpoint, and the function should return a promise holding the api response.

Then I import those static functions in the rest of my code and I pass them to `useSnapshot` in order to turn them into hot potatoes.

Reply:
🥔